test/python/circuit/library/test_iqp.py

```diff
@@ -49,9 +60,20 @@ def test_iqp(self):
         simulated = Operator(circuit)
         self.assertTrue(expected.equiv(simulated))
 
-    def test_iqp_bad(self):
+    @data(True, False)
+    def test_iqp_bad(self, use_function):
         """Test that [0,..,n-1] permutation is required (no -1 for last element)."""
-        self.assertRaises(CircuitError, IQP, [[6, 5], [2, 4]])
+        self.assertRaises(CircuitError, iqp if use_function else IQP, [[6, 5], [2, 4]])
+
+    def test_random_iqp(self):
+        """Test generating a random IQP circuit."""
+
```

Comment:
perhaps also check that the interactions are symmetric?

Reply:
I see, so in fact it could be an upper triangular matrix, since the gates are symmetric ?